### PR TITLE
CPP-990 Add brief section on OpenSSL 3.x to building doc

### PR DESCRIPTION
**Pull request by absurdfarce**

### CODE REVIEW

File: topics/building/README.md [hunk 1]
```diff
@@ -42,6 +42,16 @@ __\*\*\*__ Use the `CASS_USE_ZLIB` CMake option to enable/disable zlib support.
...
+### A Brief Note on OpenSSL 3.x
+
+Migrating from OpenSSL 1.1.x to 3.x largely involves avoiding the use of many functions which are now deprecated (consult
+the [migration guide] for much more detail on this point).  The driver does not use any of these functions so we expect
+the transition to OpenSSL 3.x to be relatively painless.  We will also note that two officially supported platforms
+(Ubuntu 22.04 and Rocky Linux 9.2) come with OpenSSL 3.x by default and the unit and integration tests all pass on these
+platforms.
+
+Given these results we expect the driver to behave well with OpenSSL 3.x.
+
```
Comment (absurdfarce) [inline topics/building/README.md:54]: @emeliawilkinson24 and @weideng1: I'm not 💯 convinced this paragraph is actually useful.  The same information is provided in more detail in the comments of [CPP-990](https://datastax-oss.atlassian.net/browse/CPP-990) and I'm not sure a paragraph like the one above gives the user any tools to make their experience with the driver better.

**Review by weideng1 — APPROVED** [review]
Good explanation of the background with the additional paragraph you added. LGTM

**Review by emeliawilkinson24 — CHANGES_REQUESTED** [review]

File: topics/building/README.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ The C/C++ driver depends on the following software:
 * [CMake] v2.8.12+
 * [libuv] 1.x
 * Kerberos v5 ([Heimdal] or [MIT]) \*
-* [OpenSSL] v1.0.x or v1.1.x \*\*
+* [OpenSSL] v1.0.x, v1.1.x or v3.x \*\*
```
[the remark this answers is not in the document]
Comment (absurdfarce) [reply topics/building/README.md:30]: Wait, are we gonna have to duke it out about the Harvard comma now? :)

File: topics/building/README.md [hunk 3]
```diff
@@ -42,6 +42,16 @@ __\*\*\*__ Use the `CASS_USE_ZLIB` CMake option to enable/disable zlib support.
            Disabling this option will disable DataStax Astra support
            within the driver; defaults to `On`.
 
+### A Brief Note on OpenSSL 3.x
+
+Migrating from OpenSSL 1.1.x to 3.x largely involves avoiding the use of many functions which are now deprecated (consult
```
[the remark this answers is not in the document]
Comment (absurdfarce) [reply topics/building/README.md:47]: I don't think this one quite works.  "Migrating from OpenSSL X to Y avoids using many functions which are now deprecated" seems wrong.  The intent is to communicate what migration involves, not the effect of migration.

File: topics/building/README.md (hunk 1 above)
Comment (absurdfarce) [reply topics/building/README.md:54]: That makes sense to me @emeliawilkinson24 .  After the other edits you had on your review I _think_ we can get to something I can live with if I just remove that last sentence.  I'll give it a shot and see what it looks like.

Comment (absurdfarce) [reply topics/building/README.md:51]: Okay, I liked how things look with that sentence out.  Take another pass when you have some time @emeliawilkinson24 and let me know what you think!

**Review by emeliawilkinson24 — APPROVED** [review]
LGTM

**Pull request merged**